crates/sui-graphql-rpc/src/server/sui_sdk_data_provider.rs

```diff
@@ -311,6 +312,20 @@ pub(crate) async fn convert_to_epoch(cl: &SuiClient, gcs: &NativeGasCostSummary)
     let gas_summary = convert_to_gas_cost_summary(gcs)?;
     let active_validators = convert_to_validators(system_state.active_validators)?;
 
+    let start_timestamp = ChronoUtc
+        .timestamp_millis_opt(system_state.epoch_start_timestamp_ms as i64)
```

Comment:
The `as` can technically also panic (although this would be unlikely outside of adverserial user input), `i64::try_from(system_state.epoch_start_timestamp_ms)` is the panic-safe way to do this (and you can produce an internal error if this fails as well).

Reply:
`as` actually can't ever panic